pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1292 @@
+    if "HT" in value:
+        output.append(value[value.index("HT") + 1].lower())
+    if "HV" in value:
+        output.append(float(value[value.index("HV") + 1][:-1]))
+    if "HV2" in value:
+        output.append(float(value[value.index("HV2") + 1][:-1]))
+    return output
+
+
+def _math_define_validator(value, values):
+    """
+    Validate the input of the math_define property
+    :param value: input parameters as a 3-element tuple
+    :param values: allowed space for each parameter
+    """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 3:
+        raise ValueError('Number of parameters {} different from 3'.format(len(value)))
+    output = (sanitize_source(value[0]), value[1], sanitize_source(value[2]))
+    for i in range(3):
+        strict_discrete_set(output[i], values=values[i])
+    return output
+
+
+def _measure_delay_validator(value, values):
+    """
+    Validate the input of the measure_delay property
+    :param value: input parameters as a 3-element tuple
+    :param values: allowed space for each parameter
+    """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 3:
+        raise ValueError('Number of parameters {} different from 3'.format(len(value)))
+    output = (value[0], sanitize_source(value[1]), sanitize_source(value[2]))
+    if output[1][0] > output[2][0]:
+        raise ValueError(f'First channel number {output[1]} must be <= than second one {output[2]}')
+    for i in range(3):
+        strict_discrete_set(output[i], values=values[i])
+    return output
+
+
+def _intensity_validator(value, values):
+    """ Validate the input of the intensity property (grid intensity and trace intensity)
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    for i in range(2):
+        strict_discrete_range(value=value[i], values=values[i], step=1)
+    return value
+
+
+class _ChunkResizer:
+    """ The only purpose of this class is to resize the chunk size of the instrument adapter.
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+    Note.
+    Only if the new chunk size is bigger than the current chunk size, it is resized. """
+
+    def __init__(self, adapter, chunk_size):
+        """ Just initialize the object attributes.
+        :param: adapter of the instrument. This is usually accessed through the
+        Instrument::adapter attribute.
+        :chunk_size: new chunk size (int).
+        """
```

Comment:
```suggestion
class _ChunkResizer:
    """ The only purpose of this class is to resize the chunk size of the instrument adapter.

    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
    waveforms.

    .. Note::
        Only if the new chunk size is bigger than the current chunk size, it is resized.

    :param adapter: adapter attribute of the instrument. This is usually accessed through the
        Instrument::adapter attribute.
    :param int chunk_size: new chunk size    
    """

    def __init__(self, adapter, chunk_size):
```
This should be commit-ready (although I did not test the docs build)

Reply:
Thank you for the suggestion.